src/app.py

```diff
@@ -2737,3 +2853,67 @@ def replace_alias(service):
             raise YunohostValidationError("dpkg_is_broken")
         elif when == "post":
             raise YunohostError("this_action_broke_dpkg")
+
+
+def _filter_and_hydrate_notifications(notifications, current_version=None, data={}):
+    return {
+        # Should we render the markdown maybe? idk
+        name: _hydrate_app_template(_value_for_locale(content_per_lang), data)
+        for name, content_per_lang in notifications.items()
+        if current_version is None
+        or name == "main"
+        or version.parse(name) > current_version
+    }
+
+
+def _display_notifications(notifications, force=False):
+    if not notifications:
+        return
+
+    for name, content in notifications.items():
+        print(f"========== {name}")
+        print(content)
+    print("==========")
+
+    _ask_confirmation("confirm_notifications_read", kind="simple", force=force)
+
+
+# FIXME: move this to Moulinette
+def _ask_confirmation(
+    question: str,
+    params: dict = {},
+    kind: str = "hard",
+    force: bool = False,
+):
+    """
+    Ask confirmation
+
+    Keyword argument:
+        question -- m18n key or string
+        params -- dict of values passed to the string formating
+        kind -- "hard": ask with "Yes, I understand", "soft": "Y/N", "simple": "press enter"
+        force -- Will not ask for confirmation
+
+    """
+    if force or Moulinette.interface.type == "api":
+        return
+
```

Comment:
(This bit is pretty important because in many context apps may be installed in a non-interactive context and you don't want to miserably crash because of this)